src/apm.js

```diff
@@ -0,0 +1,57 @@
+import { window } from './utils'
+
+function isFloat(n) {
+    return Number(n) === n && n % 1 !== 0
+}
+
+export function optimisePerformanceData(performanceData) {
+    // performance data is an object of arrays of PerformanceEntry objects
+    for (const key in performanceData) {
+        performanceData[key].forEach((performanceEntry, index) => {
+            for (const performanceEntryItemKey in performanceEntry) {
+                if (
+                    isFloat(performanceEntry[performanceEntryItemKey]) &&
+                    performanceEntry[performanceEntryItemKey].toString().match(/^\d+\.\d{4,}$/)
+                ) {
+                    performanceData[key][index][performanceEntryItemKey] = Number(
+                        performanceEntry[performanceEntryItemKey].toFixed(3)
+                    )
+                }
+
+                if (
+                    ['serverTiming', 'workerTiming'].includes(performanceEntryItemKey) &&
+                    performanceEntry[performanceEntryItemKey].length === 0
+                ) {
+                    delete performanceData[key][index][performanceEntryItemKey]
+                }
+
+                if (
+                    performanceEntryItemKey === 'entryType' &&
+                    performanceEntry[performanceEntryItemKey] === 'resource'
+                ) {
+                    delete performanceData[key][index][performanceEntryItemKey]
+                }
+
+                if (performanceEntryItemKey === 'nextHopProtocol') {
+                    delete performanceData[key][index][performanceEntryItemKey]
+                }
+
+                if (performanceEntry[performanceEntryItemKey] === 0) {
+                    delete performanceData[key][index][performanceEntryItemKey]
+                }
+            }
+        })
+    }
+
+    return performanceData
+}
```

Comment:
Somehow, these 4 identical delete statements make me think of end user JS bundle sizes. We are anyway [way beyond reasonable](https://bundlephobia.com/package/posthog-js@1.17.0), so some extra code here makes no difference (but does improve clarity of intent), and we also minify the JS in the end where variables get condensed... so not sure why I'm even bringing it up 😅